neo/io/proxyobjects.py

```diff
@@ -34,8 +34,9 @@ def __init__(self, array_annotations=None, **annotations):
             annotations['file_origin'] = str(self._rawio.source_name())
 
         # clean annotations that are not 1D
-        # TODO change array_annotations handling to avoid that
-        array_annotations = {k: v for k, v in array_annotations.items() if v.ndim==1}
+        # TODO remove this once multi-dimensional array_annotations are possible
+        array_annotations = {k: v for k, v in array_annotations.items()
+                             if (not hasattr(v, 'ndim') or v.ndim == 1)}
```

Comment:
why this ```hasattr(v, 'ndim')``` ?
I think it is better to have a np.asarray() somewhere at the good place.

Reply:
Because I don't think this is the place a general type conversion should be happening here but rather centrally in a kind of `check_array_annotations()` similar to the `check_annotations` function. I guess we will introduce this when extending the `array_annotations` to 2D. Do you agree?

Reply:
I agree for the function but having this `not hasattr(v, 'ndim')` means that we accept list as array_annotations which is my mind is not the case. It is a numpy array that have numpy indexing.

Reply:
We also accepted lists as array annotations before and shouldn't this type of conversion & consistency checks be part of the `array_annotate` methods?

Reply:
OK.
I think we should froce array annotations to be array.
here at the rawio level I think this is good that all reader render array_annotations.

Reply:
Ok, I added the conversion here. Let's see if the checks pass.